app/questionnaire/navigator.py

```diff
@@ -1,193 +1,288 @@
+        """
+        Updates the answer store and dependant location_path to the supplied answer_store
+        :param answer_store:
+        :return:
+        """
+        self.answer_store = answer_store
+        self.location_path = self.get_location_path()
+
+    def get_routing_path(self):
         """
         Returns a list of the block ids visited based on answers provided
         :return: List of block ids
         """
-        answers = answers or {}
         routing_path = []
+        blocks = self.get_blocks()
 
-        return build_path(self.blocks, self.first_block_id, answers, routing_path)
+        return self.build_path(blocks, self.first_group_id, 0, self.first_block_id, routing_path)
 
-    def can_reach_summary(self, answers=None, routing_path=None):
+    def can_reach_summary(self, routing_path=None):
         """
         Determines whether the end of a given routing path can be reached given
         a set of answers
         :param routing_path:
-        :param answers:
         :return:
         """
-        answers = answers or {}
-        routing_path = routing_path or build_path(self.blocks, self.first_block_id, answers, [])
-        last_routing_block_id = routing_path[-1]
+        blocks = self.get_blocks()
+        routing_path = routing_path or self.build_path(blocks, self.first_group_id, 0, self.first_block_id, [])
+        last_routing_block_id = routing_path[-1]['block_id']
+        last_block_id = blocks[-1]['block']['id']
 
-        if self.blocks[-1]['id'] == last_routing_block_id:
+        if last_block_id == last_routing_block_id:
             return True
 
-        routing_block_id_index = next(index for (index, b) in enumerate(self.blocks) if b["id"] == last_routing_block_id)
+        routing_block_id_index = next(index for (index, b) in enumerate(blocks) if b['block']["id"] == last_routing_block_id)
 
-        last_routing_block = self.blocks[routing_block_id_index]
+        last_routing_block = blocks[routing_block_id_index]['block']
 
         if 'routing_rules' in last_routing_block:
             for rule in last_routing_block['routing_rules']:
                 goto_rule = rule['goto']
                 if 'id' in goto_rule.keys() and goto_rule['id'] == 'summary':
-                    if 'when' in goto_rule.keys():
-                        answer_index = goto_rule['when']['id']
-                        if answer_index in answers:
-                            answer = answers[answer_index]
-                            if evaluate_rule(goto_rule, answer):
-                                return True
-                    else:
-                        return True
+                    return evaluate_goto(goto_rule, self.answer_store, 0)
         return False
 
-    def get_location_path(self, answers=None):
+    def get_location_path(self):
         """
         Returns a list of url locations visited based on answers provided
-        :param answers:
         :return:
```

Comment:
I think it would be worthwhile adding to the docstring that this returns the routing_path with the preceding and closing interstitial pages added.